shakytweaks/src/main/java/com/mindsea/shakytweaks/TweakValueResolver.kt

```diff
@@ -111,4 +122,126 @@ internal class TweakValueResolver(
             else -> throw UnsupportedOperationException("Tweak $tweak doesn't support increment")
         }
     }
+
+    fun allowsDecimals(key: String): Boolean {
+        return when (val tweak = tweakProvider.findTweak(key)) {
+            is Tweak.IntTweak -> false
+            is Tweak.LongTweak -> false
+            is Tweak.FloatTweak -> true
+            is Tweak.DoubleTweak -> true
+            else -> throw UnsupportedOperationException("Tweak $tweak isn't a number")
+        }
+    }
+
+    fun allowsNegativeNumbers(key: String): Boolean {
+        return when (val tweak = tweakProvider.findTweak(key)) {
+            is Tweak.IntTweak -> tweak.minValue < 0
+            is Tweak.LongTweak -> tweak.minValue < 0
+            is Tweak.FloatTweak -> tweak.minValue < 0
+            is Tweak.DoubleTweak -> tweak.minValue < 0
+            else -> throw UnsupportedOperationException("Tweak $tweak isn't a number")
+        }
+    }
+
+    fun updateValueFromString(key: String, string: String) {
+        when (val tweak = tweakProvider.findTweak(key)) {
+            is Tweak.IntTweak -> {
+                val updatedValue = string.toInt()
+                when {
+                    updatedValue < tweak.minValue -> {
+                        throw TweakValueOutOfRangeException(
+                            "The value $updatedValue is smaller than the minimum value (${tweak.minValue})",
```

Comment:
This message could have been built in the constructor of `TweakValueOutOfRangeException`